api/task_delete.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+// deleteTask deletes an existing task and its events. Does not delete
+// if the task is active.
+func (h *taskHandler) deleteTask(w http.ResponseWriter, r *http.Request) {
```

Comment:
Would it be better to replace this function signature with the (h *TaskLifeCycleHandler) DeleteTaskByName(w http.ResponseWriter, r *http.Request, name string) method? Then you wouldn't need a separate call out from that method, and the name string will already be parsed for you.

Reply:
You also wouldn't need the version either anymore, name would parse out the name regardless of the endpoint version

Reply:
Yeah, I was originally keeping it separate so that I could test it with the old handler and then easily switch it to the new handler. It doesn't feel all that necessary anymore to have it work for both, so I'll get rid of the DeleteTaskByName in task.go and replace it with this.